cli/src/cmd/run.rs

```diff
@@ -77,27 +106,93 @@ impl Cmd for RunArgs {
             ContractRunner::new(&mut evm, &abi, addr, Some(self.evm_opts.sender), &logs);
 
         // 5. run the test function
-        let result = runner.run_test(&func, false, None)?;
+        let result = runner.run_test(&func, false, Some(&known_contracts))?;
+
+        if self.evm_opts.debug {
+            // 6. Boot up debugger
+            let source_code: BTreeMap<u32, String> = sources
+                .iter()
+                .map(|(id, path)| {
+                    (
+                        *id,
+                        std::fs::read_to_string(path)
+                            .expect("Something went wrong reading the file"),
+                    )
+                })
+                .collect();
+
+            println!("{:?}", source_code);
+
+            let calls = evm.debug_calls();
+            println!("debugging {}", calls.len());
+            let mut flattened = Vec::new();
+            calls[0].flatten(0, &mut flattened);
+            flattened = flattened[1..].to_vec();
```

Comment:
Eh its not *necessary*. its done this way currently because we instantiate an empty entry node in the arena, as though it could run code prior to executing the first call (i.e. if depth 0/tx sender were a smart contract account). No layer 2s currently have this enabled but optimism used to obviously. I am inclined to keep the arena agnostic to depth 0 code execution and adjusting flatten or just leaving it